Src/Plugins/Security/SEC101_050.IdentifiableNpmAuthorTokenValidator.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+using Microsoft.RE2.Managed;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class IdentifiableNpmAuthorTokenValidator : DynamicValidatorBase
```

Comment:
>IdentifiableNpmAuthorTokenValidator [](http://example.com/codeflow?start=17&length=35)

create a helper that has:
- uri
- checkinformation
- dynamic validation logic
- internal classes

change this validator + npmtokenvalidator to use that helper

this will reduce the code duplication